CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_ActivityChainSpawner.uc

```diff
@@ -292,25 +292,47 @@ static function SpawnCounterDarkEvents (XComGameState NewGameState)
 	RegionRefs = GetContactedRegions();
 	DarkEventRefs = AlienHQ.ChosenDarkEvents;
 
+	`CI_Log("Initial Events:" @ DarkEventRefs.Length);
+	
 	for (i = 0; i < DarkEventRefs.Length; i++)
 	{
-		// No regions left to assign DEs, skip the rest of them
-		if (RegionRefs.Length == 0) break;
+		DarkEventRef = DarkEventRefs[i];
+		DarkEventState = XComGameState_DarkEvent(`XCOMHISTORY.GetGameStateForObjectID(DarkEventRef.ObjectID));
 		
-		if (DarkEventRefs.Length == 0)
+		`CI_Log("Evaluating:" @ DarkEventState.GetMyTemplateName());
+
+		if (DarkEventState == none)
 		{
-			`RedScreen("CI: Dark Events array empty when it should be filled");
-			break;
+			DarkEventRefs.RemoveItem(DarkEventRef);
+			i--;
```

Comment:
Do not mix `RemoveItem` and index-based iteration. It can lead to subtle bugs in case the same item appears multiple times in the array.